### PR TITLE
Add Lomb-Scargle model inspection methods

### PR DESCRIPTION
**Pull request by jakevdp**
The ``LombScargle`` methods currently lack easy access to the maximum likelihood model parameters. This PR adds model inspection methods to the ``LombScargle`` class, adds a comprehensive set of unit tests for these, and adds a section to the lomb-scargle documentation outlining how to understand these methods.

Edit: the motivation for this change was [this StackOverflow answer](https://stackoverflow.com/a/54523102/2937831), which made me realize that this functionality was missing.

### CODE REVIEW

**Comment by jakevdp** [comment]
The updated documentation when rendered looks like this:
screen shot 2019-02-05 at 1 01 02 pm

**Comment by jakevdp** [comment]
> Error seems real:

Indeed. Should be fixed now

File: astropy/stats/lombscargle/tests/test_lombscargle.py [hunk 1]
```diff
@@ -357,3 +374,51 @@ def test_autopower(data):
...
+    model = ls.model(tfit, frequency)
+    params = ls.model_parameters(frequency)
+    design = ls.design_matrix(frequency, t=tfit)
+    offset = ls.offset()
+
+    assert len(params) == int(fit_mean) + 2 * nterms
+
+    print(offset)
+    print(design)
+    print(params)
```
Comment (larrybradley) [inline astropy/stats/lombscargle/tests/test_lombscargle.py]: These `print` calls should be removed (debugging leftovers?).

Comment (jakevdp) [reply astropy/stats/lombscargle/tests/test_lombscargle.py]: Yep, thanks.

**Pull request merged**